### ISSUE TITLE
Support unsigned type

### ISSUE THREAD

**Issue by jerry73204** [chyh1990/yaml-rust#128]
I found `.as_u64()` is not available along with `.as_i64()`. It leaves me a bit inconvenience to parse the values manually. Is there a reason that it's included or is it not supported yet?

**Comment by chyh1990** [comment]
as YAML only have Integer (signed) type, so it should be parse as i64

**Comment by jerry73204** [comment]
Yes, I agree the idea to choose more a _generalised_ type, but I think we can make things further.

The [YAML spec](https://yaml.org/spec/1.2/spec.html#id2803828) states integer is arbitrary sized finite, so in I think there's no size limit and sign concept in mind. The spec later states it is parser's responsibility to reject or accept to convert to language type.

IMO, allowing `{i,u}{8,16,32,64}` basically does not violate the specification. We can simply move it `ParseError` whenever the Rust type cannot accept the value, and that would let users write less code with `?` mark.
